### ISSUE TITLE
Handling new line in strings

### ISSUE THREAD

**Issue by jeromesimeon** [accordproject/ergo#606]
New lines are not processed properly in strings.

1. New lines should be escaped when parsing. Currently:

```
ergo$ return "Hello


"
Response. "Hello


" : String
```
instead of:
```
ergo$ return "Hello


"
Response. "Hello\n\n" : String
```

This is in fact a multi-pronged problem:
- [x] Escaped carriage return, new line, backspace, double quote, tab should be parsed into their corresponding characters
- [x] carriage return, new line, backspace, double quote, tab should be printed in REPL output as escaped
- [x] JavaScript backend should escape string constant appropriately
- [ ] Printing out JSON results from `ergorun` should follow JSON escaping conventions (See https://github.com/Javascipt/Jsome/issues/17)

**Comment by jeromesimeon** [comment]
This is in fact a multi-pronged problem:
-[ ] Escaped carriage return, new line, backspace, double quote, tab should be parsed into their corresponding characters
-[ ] carriage return, new line, backspace, double quote, tab should be printed in REPL output as escaped
-[ ] JavaScript backend should escape string constant appropriately
-[ ] Printing out JSON results from `ergorun` should follow JSON escaping conventions (See https://github.com/Javascipt/Jsome/issues/17)

**Comment by jeromesimeon** [comment]
Most of it (except for Jsome bug) fixed in #607

**Comment by jeromesimeon** [comment]
Considered fixed
